spec/avromatic/model_registry_spec.rb

```diff
@@ -19,9 +18,24 @@
     end
   end
 
-  context "without a namespace prefix to remove" do
-    let(:instance) { described_class.new }
+  describe "#register_if_missing" do
+    context "for a model that has not been registered" do
+      it "registers the model" do
+        instance.register_if_missing(model)
+        expect(instance.registered?('test.nested_record')).to eql(true)
+      end
+    end
+
+    context "for model that has been registered" do
+      before { instance.register(model) }
 
+      it "does not raise an error" do
```

Comment:
Should we assert that the "old" version of the model is still registered (which may require some mocking)?